lib/src/solver/version_solver.dart

```diff
@@ -431,7 +431,9 @@ class VersionSolver {
       List<PackageId> packages) async {
     var availableVersions = <String, List<Version>>{};
     for (var package in packages) {
-      var cached = _packageListers[package.toRef()]?.cachedVersions;
+      var packageLister = _packageListers[package.toRef()];
```

Comment:
I get a feeling we should never ask the packageLister here.

The sources will cache the available versions themselves.

This might be for another PR...

Reply:
Noted for another PR